pkg/rehearse/configmaps.go

```diff
@@ -91,48 +139,34 @@ func (c *CMManager) createCM(name string, data []updateconfig.ConfigMapUpdate) e
 	return nil
 }
 
-func genChanges(root string, sources []config.ConfigMapSource) ([]prowgithub.PullRequestChange, error) {
+func genChanges(root string, patterns sets.String) ([]prowgithub.PullRequestChange, error) {
 	var ret []prowgithub.PullRequestChange
-	for _, f := range sources {
-		err := filepath.Walk(filepath.Join(root, f.PathInRepo), func(path string, info os.FileInfo, err error) error {
-			if err != nil || info.IsDir() {
-				return err
-			}
-			// Failure is impossible per filepath.Walk's API.
-			path, err = filepath.Rel(root, path)
+	err := filepath.Walk(root, func(path string, info os.FileInfo, err error) error {
+		if err != nil || info.IsDir() {
+			return err
+		}
+		// Failure is impossible per filepath.Walk's API.
+		path, err = filepath.Rel(root, path)
+		if err != nil {
+			return err
+		}
+		for pattern := range patterns {
+			match, err := zglob.Match(pattern, path)
 			if err != nil {
 				return err
 			}
-			ret = append(ret, prowgithub.PullRequestChange{
-				Filename: path,
-				Status:   string(prowgithub.PullRequestFileModified),
-			})
-			return nil
-		})
-		if err != nil {
-			return nil, err
-		}
-	}
-	return ret, nil
-}
-
-func (c *CMManager) validateChanges(changes []prowgithub.PullRequestChange) error {
```

Comment:
This validation is no longer needed b/c the input the the `Create()` method is now a `ConfigMaps` instance, which cannot be successfully populated by `NewConfigMaps` if any of the changed files are not covered by a `config-updater` pattern.